### ISSUE TITLE
Array of objects parsing

### ISSUE THREAD

**Issue by dmitriyles** [hapijs/qs#47]
I'm trying to parse an IPN message from PayPal, and it has an array of objects in a query string:
```url
transaction%5B0%5D.is_primary_receiver=true&transaction%5B0%5D.id_for_sender_txn=5GN104358V194442M&log_default_shipping_address_in_transaction=false&transaction%5B0%5D.receiver=seller1.test%40mercher.net&action_type=PAY&ipn_notification_url=http%3A//staging.mercherdev.com/ipn&transaction%5B1%5D.paymentType=SERVICE&transaction%5B0%5D.amount=USD+136.25&charset=windows-1252&transaction_type=Adaptive+Payment+PAY&transaction%5B1%5D.id_for_sender_txn=9L346159YT798301J&transaction%5B1%5D.is_primary_receiver=false&transaction%5B0%5D.status=Completed&notify_version=UNVERSIONED&transaction%5B0%5D.id=3D429284U5773243B&cancel_url=http%3A//staging.mercherdev.com/test&transaction%5B1%5D.status_for_sender_txn=Completed&transaction%5B1%5D.receiver=dmitriy.s.les-facilitator%40gmail.com&verify_sign=AFcWxV21C7fd0v3bYYYRCpSSRl31Aaz5mmivhYQQvF-Zz1csgdIo0LYX&sender_email=buyer1.test%40mercher.net&fees_payer=PRIMARYRECEIVER&transaction%5B0%5D.status_for_sender_txn=Completed&return_url=http%3A//staging.mercherdev.com/test&transaction%5B0%5D.paymentType=GOODS&transaction%5B1%5D.amount=USD+2.50&reverse_all_parallel_payments_on_error=true&tracking_id=32&transaction%5B1%5D.pending_reason=NONE&pay_key=AP-8LF09716FJ765462U&transaction%5B1%5D.id=43253900X5221472M&transaction%5B0%5D.pending_reason=NONE&status=COMPLETED&transaction%5B1%5D.status=Completed&test_ipn=1&payment_request_date=Mon+Oct+06+09%3A16%3A01+PDT+2014
```
`Qs.parse()` gives me this result:
```json
{
    "transaction":                                 [ "NONE", "Completed" ],
    "log_default_shipping_address_in_transaction": "false",
    "action_type":                                 "PAY",
    "ipn_notification_url":                        "http://staging.mercherdev.com/ipn",
    "charset":                                     "windows-1252",
    "transaction_type":                            "Adaptive Payment PAY",
    "notify_version":                              "UNVERSIONED",
    "cancel_url":                                  "http://staging.mercherdev.com/test",
    "verify_sign":                                 "AFcWxV21C7fd0v3bYYYRCpSSRl31Aaz5mmivhYQQvF-Zz1csgdIo0LYX",
    "sender_email":                                "buyer1.test@mercher.net",
    "fees_payer":                                  "PRIMARYRECEIVER",
    "return_url":                                  "http://staging.mercherdev.com/test",
    "reverse_all_parallel_payments_on_error":      "true",
    "tracking_id":                                 "32",
    "pay_key":                                     "AP-8LF09716FJ765462U",
    "status":                                      "COMPLETED",
    "test_ipn":                                    "1",
    "payment_request_date":                        "Mon Oct 06 09:16:01 PDT 2014"
}
```
while I was expecting for:
```json
{
    "transaction":                                 [
        {
            "is_primary_receiver":   "true",
            "id_for_sender_txn":     "1EN35283R74763703",
            "receiver":              "seller1.test@mercher.net",
            "amount":                "USD 107.00",
            "status":                "Completed",
            "id":                    "90E58532R7615693H",
            "status_for_sender_txn": "Completed",
            "paymentType":           "GOODS",
            "pending_reason":        "NONE"
        },
        {
            "paymentType":           "SERVICE",
            "id_for_sender_txn":     "7XF42668MS476925V",
            "is_primary_receiver":   "false",
            "status_for_sender_txn": "Completed",
            "receiver":              "dmitriy.s.les-facilitator@gmail.com",
            "amount":                "USD 2.00",
            "pending_reason":        "NONE",
            "id":                    "6XR12273235492049",
            "status":                "Completed"
        }
    ],
    "log_default_shipping_address_in_transaction": "false",
    "action_type":                                 "PAY",
    "ipn_notification_url":                        "http://staging.mercherdev.com/ipn",
    "charset":                                     "windows-1252",
    "transaction_type":                            "Adaptive Payment PAY",
    "notify_version":                              "UNVERSIONED",
    "cancel_url":                                  "http://staging.mercherdev.com/test",
    "verify_sign":                                 "AFcWxV21C7fd0v3bYYYRCpSSRl31AW50ZrxhpTVKQjEObfEFfTcdVouU",
    "sender_email":                                "buyer1.test@mercher.net",
    "fees_payer":                                  "PRIMARYRECEIVER",
    "return_url":                                  "http://staging.mercherdev.com/test",
    "reverse_all_parallel_payments_on_error":      "true",
    "tracking_id":                                 "31",
    "pay_key":                                     "AP-10U799054T2106106",
    "status":                                      "COMPLETED",
    "test_ipn":                                    "1",
    "payment_request_date":                        "Mon Oct 06 09:00:28 PDT 2014"
}
```
Is there any way to achieve this? I'm using v2.2.4

**Comment by nlf** [comment]
Closed via #66

**Comment by hueniverse** [comment]
So... this is not marked with a release, is actually a breaking change (see https://github.com/hapijs/hapi/issues/2718) and the comment above is referencing back this issue.

This is a pretty big change. I am not sure what to do about this now as it is breaking code for hapi users.
